daml/ContingentClaims/Lifecycle.daml

```diff
@@ -42,14 +42,20 @@ data Result t a = Result with
 -- Results in a function taking today's date, and returning the pruned tree + pending settlements.
 lifecycle : (Ord t, Eq a, CanAbort m)
   => (a -> t -> m Decimal)
+  -- ^ function to evaluate observables 
+  -> t
+  -- ^ start date of the input claim
   -> C t a
-  -> t -> m (Result t a)
-lifecycle spot
+  -- ^ the input claim
+  -> t
+  -- ^ the today's date
+  -> m (Result t a)
+lifecycle spot startDate
   = runKleisli
   . fmap (uncurry $ flip Result)
   . runWriterT
   . apoCataM pruneZeros' acquireThenSettle
-  . (1.0, )
+  . ((startDate, 1.0), )
```

Comment:
So I'm still not 100% convinced that this is different from `t`. I'll need some time to think it over.